### PR TITLE
get vm by name instead of search in a list

### PR DESCRIPTION
**Pull request by xunpan**
**What this PR does / why we need it**:
make code logic simple and clear. 

**Specify your PR reviewers**:

/cc @gyliu513 @jichenjc 

**Special notes for your reviewer**:
Still need to do below things in this PR before review
 - [x] cover domain filter
 - [x] test function

### CODE REVIEW

File: pkg/cloud/ibmcloud/clients/machines.go [hunk 1]
```diff
@@ -176,32 +177,30 @@ func (gs *GuestService) DeleteGuest(Id int) error {
...
-	for _, guest := range guests {
-		// FIXME: how to unique identify one guest
-		if *guest.Hostname == name {
-			klog.Infof("Found guest with ID %d for %q", *guest.Id, name)
-			return &guest, nil
-		}
+	if len(guests) > 1 {
+		// I noticed that IBM Cloud can use same name for 2 machines.
+		// It is bad for our case. Print a message to make it to be noticed.
+		klog.Errorf("Getting more than one virtual guest. The first is used.")
```
Comment (gyliu513) [inline pkg/cloud/ibmcloud/clients/machines.go]: Print out the guest name here for better debug.

File: pkg/cloud/ibmcloud/clients/machines.go [hunk 2]
```diff
@@ -176,32 +177,32 @@ func (gs *GuestService) DeleteGuest(Id int) error {
...
-	for _, guest := range guests {
-		// FIXME: how to unique identify one guest
-		if *guest.Hostname == name {
-			klog.Infof("Found guest with ID %d for %q", *guest.Id, name)
-			return &guest, nil
-		}
+	if len(guests) > 1 {
+		// I noticed that IBM Cloud can use same name for 2 machines.
+		// It is bad for our case. Print a message to make it to be noticed.
+		klog.Errorf("Getting more than one virtual guests by filter (hostname=%s, domain=%s): %v. The first is used.",
```
Comment (gyliu513) [inline pkg/cloud/ibmcloud/clients/machines.go]: How about `The first host %s is used`?

File: pkg/cloud/ibmcloud/clients/machines.go (hunk 1 above)
Comment (xunpan) [reply pkg/cloud/ibmcloud/clients/machines.go]: I use guest ID because names are the same when issue happens.

**Comment by xunpan** [comment]
addressed comments and rebase done

@gyliu513

File: pkg/cloud/ibmcloud/clients/machines.go (hunk 2 above)
Comment (xunpan) [reply pkg/cloud/ibmcloud/clients/machines.go]: I use guest ID because names are the same when issue happens.

**Pull request merged**